core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -624,6 +624,49 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
     assert(store.getRemoteBytes("list1").isEmpty)
   }
 
+  Seq(false, true).foreach { stream =>
+    test(s"test for Block replication retry logic (stream = $stream)") {
+      // Retry replication logic for 2 failures
+      conf.set(STORAGE_MAX_REPLICATION_FAILURE, 2)
+      // Custom block replication policy which prioritizes BlockManagers as per hostnames
+      conf.set(STORAGE_REPLICATION_POLICY, classOf[SortOnHostNameBlockReplicationPolicy].getName)
+      // To use upload block stream flow, set maxRemoteBlockSizeFetchToMem to 0
+      val maxRemoteBlockSizeFetchToMem = if (stream) 0 else Int.MaxValue - 512
+      conf.set(MAX_REMOTE_BLOCK_SIZE_FETCH_TO_MEM, maxRemoteBlockSizeFetchToMem.toLong)
+      val blockManagers = (0 to 6).map(index => makeBlockManager(7800, s"host-$index", master))
+      val a1 = new Array[Byte](4000)
+      val a2 = new Array[Byte](4000)
+      val a3 = new Array[Byte](4000)
+      // Put 4000 byte sized RDD Blocks in block manager 1, 2 and 4. So that they
+      // won't have space for another 4000 bytes block.
+      blockManagers(1).putSingle(rdd(0, 1), a1, StorageLevel.MEMORY_ONLY)
+      blockManagers(2).putSingle(rdd(0, 2), a2, StorageLevel.MEMORY_ONLY)
+      blockManagers(4).putSingle(rdd(0, 3), a3, StorageLevel.MEMORY_ONLY)
```

Comment:
Since the block is stored in memory_only mode, so it's possible that the block can be evicted by the newer block.

Actually, personally, I prefer the original way, which is setting low memory for these BlockManagers. But I do notice the flay too, though I don't look into for details.

Reply:
@Ngone51  I have replaced the old test with a new one. But the build failed with some intermittent issue. Can you please retrigger the build.